imgui/core.pyx

```diff
@@ -573,6 +573,10 @@ VIEWPORT_FLAGS_OWNED_BY_APP = enums.ImGuiViewportFlags_OwnedByApp
 
 include "imgui/common.pyx"
 
+# For objects that cimgui stores as void* (such as texture_id) but need to be kept alive for rendering.
+# The cache is cleared on new_frame().
+_keepalive_cache = []
```

Comment:
I'd suggest to store this cache in the `_ImGuiContext`, so that this will still work if there are multiple contexts